bench/Bench/Utils.hs

```diff
@@ -0,0 +1,176 @@
+  pure BenchEnv {dbFilePath, dbEnv, db}
+
+createDbDir :: MonadIO m => m FilePath
+createDbDir = do
+  sysTmpDir <- liftIO getTemporaryDirectory
+  liftIO $ createTempDirectory sysTmpDir "bench_cursor"
+
+cleanupBenchEnv :: MonadIO m => BenchEnv k v -> m ()
+cleanupBenchEnv tenv = do
+  liftIO $ closeEnvironmentBlocking (dbEnv tenv)
+  deleteDbDir (dbFilePath tenv)
+
+deleteDbDir :: MonadIO m => FilePath -> m ()
+deleteDbDir = liftIO . removeDirectoryRecursive
+
+{-------------------------------------------------------------------------------
+  Defaults for LMDB limits
+-------------------------------------------------------------------------------}
+
+simpleLMDBLimits :: Limits
+simpleLMDBLimits = defaultLimits
+
+{-------------------------------------------------------------------------------
+  Defaults for database population
+-------------------------------------------------------------------------------}
+
+-- | @'noPopulateDb' db@ does not populate the LMDB database @db@ with any
+-- entries.
+noPopulateDb :: Database k v -> Transaction ReadWrite ()
+noPopulateDb _ = pure ()
+
+-- | @'populateDbUnique' n db@ populates the LMDB database @db@ with @n@ unique
+-- key-value pairs.
+--
+-- Note: ensure that @n@ is much smaller than the range of values that are
+-- generated for @'Arbitrary' k@, or this function may not terminate within
+-- reasonable time, let alone terminate at all. We use a simple "retry if a
+-- generated key is already present in the database (i.e., a /clash/)" strategy
+-- to generate exactly @n@ unique keys, which means that clashes occur
+-- inhibitively often as @n@ approaches the size of the of the range of @k@.
+populateDbUnique ::
+     ( Serialise k, Serialise v
+     , Arbitrary k, Arbitrary v
+     , Ord k
+     )
+  => Int
+  -> Database k v
+  -> Transaction ReadWrite ()
+populateDbUnique n db
+  | n <= 0    = pure ()
+  | otherwise = do
+      k <- liftIO $ generate arbitrary
+      v <- liftIO $ generate arbitrary
+      b <- Internal.putNoOverwrite db k v
+      if b then
+        populateDbUnique (n - 1) db
+      else
+        populateDbUnique n db
+
+{-------------------------------------------------------------------------------
+  Default types for keys and values
+-------------------------------------------------------------------------------}
+
+newtype Key = Key B.ShortByteString
+  deriving stock (Show, Eq, Ord, Generic)
+  deriving newtype (NFData, Serialise)
+
+-- | Size of a @'Key'@ in number of bytes.
+keySize :: Int
+keySize = 32
+
+key0 :: Key
+key0 = Key . B.pack $ replicate keySize 0
+
+newtype Value = Value B.ShortByteString
+  deriving stock (Show, Eq, Ord, Generic)
+  deriving newtype (NFData, Serialise)
+
+-- | Size of a @'Value'@ in number of bytes.
+valueSize :: Int
+valueSize = 64
+
+-- | A value of type @'Key'@ is chosen uniformly from the entire range of the
+-- type @'Key'@: any possible @'Word8'@ vector of length @'keySize'@.
```

Comment:
Like in the previous PR, maybe it's worth saying a few words about why we should use `arbitraryBoundedIntegral` instead of simply `arbitrary`.

Reply:
I've added a comment